### ISSUE TITLE
[LabelledNumbers] Labelled numbers have confusing algebra rules

### ISSUE THREAD

**Issue by ogauthe** [ITensor/ITensors.jl#1364]
Currently the output type of a sum of `LabelledInteger` depends on parity.

```julia
using NDTensors.LabelledNumbers: LabelledInteger
n = LabelledInteger(2, "x")
println(n + n)
println(n + n + n)
println(sum([n, n]))
println(sum([n, n, n]))
```
returns either the integer `4` or `LabelledInteger{Int64, String}(6, "x")`. Also note that `n + n != 2*n`.

I think the result should be well defined: either always an `Int` or always a `LabelledInteger` + an error if labels do not match.

**Comment by mtfishman** [comment]
Good point. The reasoning was that I was trying to make algebra operations preserve labels if we could prove they could at compile time (at least in binary operations), but I get your point that the current algebra rules leads to confusing behavior once you go beyond binary operations.

I just checked that dropping labels when labelled numbers are involved in addition and multiplication doesn't break any of the `GradedAxes` tests, let's try that approach. That is the "friendliest" approach, since it allows labelled numbers to be used in a lot of generic Julia code (i.e. anywhere that other `Integer` subtypes can be used), as opposed to erroring if there are mismatched labels.

**Comment by mtfishman** [comment]
I made a PR here: #1367.

**Comment by ogauthe** [comment]
Thank you for the quick fix. I find there is still a confusing result for
```julia
sum([n])   # LabelledInteger{Int64, String}(2, "x") 
sum([n, n])  # Int
```

But it may be simpler to call `sum([n], init=0)` instead of trying to fix this.

**Comment by mtfishman** [comment]
Yeah, it seems best to use `init=0` for that, but I see the point that it makes it a bit strange in generic code where you don't know the length of the output but want it to act the same way independent of the length.

**Comment by mtfishman** [comment]
Closed by #1367.
